app/javascript/article-form/components/EditorActions.jsx

```diff
@@ -18,102 +19,91 @@ const Icon = () => (
   </svg>
 );
 
-export class EditorActions extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      moreConfigShowing: false,
-    };
-  }
-
-  setCommonProps = ({ moreConfigShowing = false }) => {
-    return {
-      moreConfigShowing,
-    };
-  };
-
-  toggleMoreConfig = (e) => {
-    const { moreConfigShowing } = this.state;
-    e.preventDefault();
-    this.setState({
-      ...this.setCommonProps({ moreConfigShowing: !moreConfigShowing }),
-    });
-  };
+export const EditorActions = ({
+  onSaveDraft,
+  onPublish,
+  onClearChanges,
+  published,
+  edited,
+  version,
+  passedData,
+  onConfigChange,
+  submitting,
+}) => {
+  const [moreConfigShowing, setMoreConfig] = useState(false);
 
-  render() {
-    const {
-      onSaveDraft,
-      onPublish,
-      onClearChanges,
-      published,
-      edited,
-      version,
-      passedData,
-      onConfigChange,
-      submitting,
-    } = this.props;
+  const v1 = version === 'v1';
+  const v2 = version === 'v2';
 
-    const { moreConfigShowing } = this.state;
+  const toggleMoreConfig = useCallback(
```

Comment:
refactor (blocking): `useCallback` seems overkill here. For context, see  [When to useMemo and useCallback](https://kentcdodds.com/blog/usememo-and-usecallback).